### ISSUE TITLE
trial --coverage makes some tests fail spuriously

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1944]
|[<img alt="washort's avatar" src="https://avatars.githubusercontent.com/u/947487?s=50" width="50" height="50">](https://github.com/washort)| @washort reported|
|-|-|
|Trac ID|trac#1944|
|Type|defect|
|Created|2006-07-21 02:39:48Z|

with Divmod SVN rev 7906 and Twisted SVN rev 17626:

```
multivac% trial --coverage nevow.test.test_later.SuperLaterDataTest.test_rendererCalledOnce               ~/Projects/Divmod/trunk

Running 1 tests.
nevow.test.test_later
  SuperLaterDataTest
    test_rendererCalledOnce ...                                          [FAIL]
                                        [ERROR]

===============================================================================
[FAIL]: nevow.test.test_later.SuperLaterDataTest.test_rendererCalledOnce

Traceback (most recent call last):
  File "/home/washort/Projects/Divmod/trunk/Nevow/nevow/test/test_later.py", line 121, in test_rendererCalledOnce
    self.assertEquals(req.v, '<html>1</html>')
twisted.trial.unittest.FailTest: '' != '<html>1</html>'
===============================================================================
[ERROR]: nevow.test.test_later.SuperLaterDataTest.test_rendererCalledOnce

Traceback (most recent call last):
  File "/home/washort/Projects/Divmod/trunk/Nevow/nevow/flat/twist.py", line 23, in _drive
    next = iterable.next()
  File "/home/washort/Projects/Divmod/trunk/Nevow/nevow/flat/ten.py", line 84, in iterflatten
    for item in gen:
  File "/home/washort/Projects/Divmod/trunk/Nevow/nevow/flat/flatstan.py", line 197, in ListSerializer
    yield serialize(item, context)
  File "/home/washort/Projects/Divmod/trunk/Nevow/nevow/flat/ten.py", line 71, in serialize
    return partialflatten(context, obj)
  File "/home/washort/Projects/Divmod/trunk/Nevow/nevow/flat/ten.py", line 62, in partialflatten
    return flattener(obj, context)
  File "/home/washort/Projects/Divmod/trunk/Nevow/nevow/flat/flatstan.py", line 260, in DirectiveSerializer
    renderer = rendererFactory.renderer(context, original.name)
exceptions.TypeError: renderer() takes exactly 2 arguments (3 given)
-------------------------------------------------------------------------------
Ran 1 tests in 0.123s

FAILED (failures=1, errors=1)
```

This test passes without --coverage.

<details><summary>Searchable metadata</summary>

```
trac-id__1944 1944
type__defect defect
reporter__washort washort
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__wontfix wontfix
component__trial trial
keywords__ 
time__1153449588000000 1153449588000000
changetime__1297657849000000 1297657849000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_1"></a>|@jerub commented|
|-|-|

I see the same thing when testing the xmpp-client-1046-2 branch:
```
$ trial --coverage twisted.words.test.test_jabberxmlstream.XmlStreamTest.testAuthenticator
Setting coverage directory to coverage.
Running 1 tests.
twisted.words.test.test_jabberxmlstream
  XmlStreamTest
    testAuthenticator ...                                               [ERROR]

===============================================================================
[ERROR]: twisted.words.test.test_jabberxmlstream.XmlStreamTest.testAuthenticator

Traceback (most recent call last):
  File "/Users/stephen/src/Twisted/trunk/twisted/words/test/test_jabberxmlstream.py", line 274, in testAuthenticator
    xs = xmlstream.XmlStream(a)
  File "/Users/stephen/src/Twisted/trunk/twisted/words/protocols/jabber/xmlstream.py", line 373, in __init__
    authenticator.associateWithStream(self)
exceptions.TypeError: 'list' object is not callable
-------------------------------------------------------------------------------
Ran 1 tests in 0.189s

FAILED (errors=1)
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set status to `closed`|
|-|-|

This is a Python bug.  See <http://bugs.python.org/issue1569356>.  It seems to be fixed in Python 2.6.  I've requested that the fix be backported to Python 2.5.  I don't think there's anything _we_ can do to fix this.  It's a bug in the runtime.

I'm closing this as _wontfix_ for lack of a better available resolution.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
